quarkchain/cluster/jsonrpc.py

```diff
@@ -258,6 +260,36 @@ def minor_block_encoder(block, include_transactions=False, extra_info=None):
     return d
 
 
+def minor_block_header_encoder(header):
+    """Encode a block header as JSON object.
+
+    :param header: a :class:`ethereum.block.Block`
+    :returns: a json encodable dictionary
+    """
+
+    d = {
+        "id": id_encoder(header.get_hash(), header.branch.get_full_shard_id()),
+        "height": quantity_encoder(header.height),
+        "hash": data_encoder(header.get_hash()),
+        "fullShardId": quantity_encoder(header.branch.get_full_shard_id()),
+        "chainId": quantity_encoder(header.branch.get_chain_id()),
+        "shardId": quantity_encoder(header.branch.get_shard_id()),
+        "hashPrevMinorBlock": data_encoder(header.hash_prev_minor_block),
+        "idPrevMinorBlock": id_encoder(
+            header.hash_prev_minor_block, header.branch.get_full_shard_id()
+        ),
+        "hashPrevRootBlock": data_encoder(header.hash_prev_root_block),
+        "nonce": quantity_encoder(header.nonce),
+        "miner": address_encoder(header.coinbase_address.serialize()),
+        "coinbase": balances_encoder(header.coinbase_amount_map),
+        "difficulty": quantity_encoder(header.difficulty),
+        "extraData": data_encoder(header.extra_data),
+        "gasLimit": quantity_encoder(header.evm_gas_limit),
+        "timestamp": quantity_encoder(header.create_time),
+    }
+    return d
```

Comment:
can reuse this code for `root_block_encoder` to encode included headers and also `minor_block_encoder`